.github/workflows/build_macos.yaml

```diff
@@ -9,14 +9,19 @@ jobs:
       matrix:
         config: [debug, release]
         compiler: [clang]
-        arch: [x86, arm64]
+        os: [13, 14, 15]
+        # arch: [x86, arm64]]
         # Check https://github.com/actions/runner-images#available-images
         include:
-          - arch: x86
-            os: 12
-          - arch: arm64
-            os: 14
-    name: MacOS ${{ matrix.config }} ${{ matrix.compiler }} ${{ matrix.arch }}
+          - os: 13
+            osname: macOS13 x86
```

Comment:
You use this for both the artifact name, which deliberately uses spaces, and for the DMG filename, which avoids spaces - but fails doing so this way. Oh, and in the matrix job name, where it duplicates "MacOS".

So I think you should just stick to the old formats, and add any extra you'd like just there.